In [4]:
import numpy as np
import pandas as pd

In [52]:
df = pd.read_csv("Iris (1).csv")
df.head(3)

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0   1            5.1           3.5            1.4           0.2  Iris-setosa
1   2            4.9           3.0            1.4           0.2  Iris-setosa
2   3            4.7           3.2            1.3           0.2  Iris-setosa

# **IRIS**

### 1 What type of data does each column in the Iris dataset contain?

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             150 non-null    int64  
 1   SepalLengthCm  150 non-null    float64
 2   SepalWidthCm   150 non-null    float64
 3   PetalLengthCm  150 non-null    float64
 4   PetalWidthCm   150 non-null    float64
 5   Species        150 non-null    object 
dtypes: float64(4), int64(1), object(1)
memory usage: 7.2+ KB


### 2 What are the dimensions of the Iris dataset?

In [ ]:
df.ndim

In [19]:
df.shape

(150, 6)

### 3 How many unique species of flowers are present in the dataset, and what are their names?

In [23]:
df["Species"].nunique()

3

In [25]:
df["Species"].unique()

array(['Iris-setosa', 'Iris-versicolor', 'Iris-virginica'], dtype=object)

### 4 Check if there are any nan values in the dataset.

In [28]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             150 non-null    int64  
 1   SepalLengthCm  150 non-null    float64
 2   SepalWidthCm   150 non-null    float64
 3   PetalLengthCm  150 non-null    float64
 4   PetalWidthCm   150 non-null    float64
 5   Species        150 non-null    object 
dtypes: float64(4), int64(1), object(1)
memory usage: 7.2+ KB


### 5 convert the "id" column from int to float.

In [60]:
df["Id"]=df["Id"].astype("float")

In [62]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             150 non-null    float64
 1   SepalLengthCm  150 non-null    float64
 2   SepalWidthCm   150 non-null    float64
 3   PetalLengthCm  150 non-null    float64
 4   PetalWidthCm   150 non-null    float64
 5   Species        150 non-null    object 
dtypes: float64(5), object(1)
memory usage: 7.2+ KB


### 6 What is the range of values for the ‘SepalLengthCm’ and ‘PetalWidthCm’?

In [74]:
range1 = (df["SepalLengthCm"].min() , df["SepalLengthCm"].max())
range1

(4.3, 7.9)

In [78]:
range2 = (df["PetalWidthCm"].min() , df["PetalWidthCm"].max())
range2

(0.1, 2.5)

### 7 What is the mean and median of the ‘PetalLengthCm’ column?

In [81]:
df["PetalLengthCm"].mean()

3.758666666666666

In [83]:
df["PetalLengthCm"].median()

4.35

### 8 What does the df.describe().T function do, and how can it help in performing exploratory data analysis on the Iris dataset? Interpret the output.`

In [90]:
df.describe().T

count       mean        std  min    25%    50%     75%    max
Id             150.0  75.500000  43.445368  1.0  38.25  75.50  112.75  150.0
SepalLengthCm  150.0   5.843333   0.828066  4.3   5.10   5.80    6.40    7.9
SepalWidthCm   150.0   3.054000   0.433594  2.0   2.80   3.00    3.30    4.4
PetalLengthCm  150.0   3.758667   1.764420  1.0   1.60   4.35    5.10    6.9
PetalWidthCm   150.0   1.198667   0.763161  0.1   0.30   1.30    1.80    2.5

##### it helps by :
- Identifying data ranges → min and max tell you the spread.
- Detecting outliers → if max/min are very far from 25%–75%, possible outliers.
- Understanding central tendency → mean vs median shows skewness.
- Comparing variability → std tells how spread out features are.
- Checking data quality → count helps verify missing values.
##### output interpretation
- SepalLengthCm :
Mean ≈ 5.8, Range ≈ 4.3–7.9
Relatively balanced around mean (median close to mean).
- SepalWidthCm:
Mean ≈ 3.05, Range ≈ 2.0–4.4
Slightly smaller spread compared to sepal length.
- PetalLengthCm:
Mean ≈ 3.76, Range ≈ 1.0–6.9
Higher variability → good feature for classification.
- PetalWidthCm:
Mean ≈ 1.2, Range ≈ 0.1–2.5
Large difference between min and max → also a strong classifier

###  9 How would you select the first five rows of the Iris dataset using iloc

In [107]:
df.iloc[0:5]

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0  1.0            5.1           3.5            1.4           0.2  Iris-setosa
1  2.0            4.9           3.0            1.4           0.2  Iris-setosa
2  3.0            4.7           3.2            1.3           0.2  Iris-setosa
3  4.0            4.6           3.1            1.5           0.2  Iris-setosa
4  5.0            5.0           3.6            1.4           0.2  Iris-setosa

### 10 How can you access the 'PetalLengthCm' column for rows where the index label is 'iris-setosa' using loc?

In [112]:
print(df.loc[df["Species"] == "Iris-setosa", "PetalLengthCm"])

0     1.4
1     1.4
2     1.3
3     1.5
4     1.4
5     1.7
6     1.4
7     1.5
8     1.4
9     1.5
10    1.5
11    1.6
12    1.4
13    1.1
14    1.2
15    1.5
16    1.3
17    1.4
18    1.7
19    1.5
20    1.7
21    1.5
22    1.0
23    1.7
24    1.9
25    1.6
26    1.6
27    1.5
28    1.4
29    1.6
30    1.6
31    1.5
32    1.5
33    1.4
34    1.5
35    1.2
36    1.3
37    1.5
38    1.3
39    1.5
40    1.3
41    1.3
42    1.3
43    1.6
44    1.9
45    1.4
46    1.6
47    1.4
48    1.5
49    1.4
Name: PetalLengthCm, dtype: float64


### 11 How  you filter the dataset to include only rows where 'SepalLengthCm' is greater than 5.0 and 'PetalWidthCm' is less than 1.5 using the & operator?

In [124]:
df[(df["SepalLengthCm"] > 5.0 ) & (df["PetalWidthCm"] < 1.5)]


Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm  \
0      1.0            5.1           3.5            1.4           0.2   
5      6.0            5.4           3.9            1.7           0.4   
10    11.0            5.4           3.7            1.5           0.2   
14    15.0            5.8           4.0            1.2           0.2   
15    16.0            5.7           4.4            1.5           0.4   
16    17.0            5.4           3.9            1.3           0.4   
17    18.0            5.1           3.5            1.4           0.3   
18    19.0            5.7           3.8            1.7           0.3   
19    20.0            5.1           3.8            1.5           0.3   
20    21.0            5.4           3.4            1.7           0.2   
21    22.0            5.1           3.7            1.5           0.4   
23    24.0            5.1           3.3            1.7           0.5   
27    28.0            5.2           3.5            1.5           0.2   
28    29.0            5.2           3.4            1.4           0.2   
31    32.0            5.4           3.4            1.5           0.4   
32    33.0            5.2           4.1            1.5           0.1   
33    34.0            5.5           4.2            1.4           0.2   
36    37.0            5.5           3.5            1.3           0.2   
39    40.0            5.1           3.4            1.5           0.2   
44    45.0            5.1           3.8            1.9           0.4   
46    47.0            5.1           3.8            1.6           0.2   
48    49.0            5.3           3.7            1.5           0.2   
50    51.0            7.0           3.2            4.7           1.4   
53    54.0            5.5           2.3            4.0           1.3   
55    56.0            5.7           2.8            4.5           1.3   
58    59.0            6.6           2.9            4.6           1.3   
59    60.0            5.2           2.7            3.9           1.4   
62    63.0            6.0           2.2            4.0           1.0   
63    64.0            6.1           2.9            4.7           1.4   
64    65.0            5.6           2.9            3.6           1.3   
65    66.0            6.7           3.1            4.4           1.4   
67    68.0            5.8           2.7            4.1           1.0   
69    70.0            5.6           2.5            3.9           1.1   
71    72.0            6.1           2.8            4.0           1.3   
73    74.0            6.1           2.8            4.7           1.2   
74    75.0            6.4           2.9            4.3           1.3   
75    76.0            6.6           3.0            4.4           1.4   
76    77.0            6.8           2.8            4.8           1.4   
79    80.0            5.7           2.6            3.5           1.0   
80    81.0            5.5           2.4            3.8           1.1   
81    82.0            5.5           2.4            3.7           1.0   
82    83.0            5.8           2.7            3.9           1.2   
87    88.0            6.3           2.3            4.4           1.3   
88    89.0            5.6           3.0            4.1           1.3   
89    90.0            5.5           2.5            4.0           1.3   
90    91.0            5.5           2.6            4.4           1.2   
91    92.0            6.1           3.0            4.6           1.4   
92    93.0            5.8           2.6            4.0           1.2   
94    95.0            5.6           2.7            4.2           1.3   
95    96.0            5.7           3.0            4.2           1.2   
96    97.0            5.7           2.9            4.2           1.3   
97    98.0            6.2           2.9            4.3           1.3   
98    99.0            5.1           2.5            3.0           1.1   
99   100.0            5.7           2.8            4.1           1.3   
134  135.0            6.1           2.6         

### 12 Using loc and & to Select Rows Labeled 'Versicolor' where 'SepalLengthCm' is Between 5.5 and 6.5

In [131]:
df[(df["Species"] == "Iris-versicolor") & (( df["SepalLengthCm"] > 5.5) & (df["SepalLengthCm"]< 6.5))]

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm  \
51   52.0            6.4           3.2            4.5           1.5   
55   56.0            5.7           2.8            4.5           1.3   
56   57.0            6.3           3.3            4.7           1.6   
61   62.0            5.9           3.0            4.2           1.5   
62   63.0            6.0           2.2            4.0           1.0   
63   64.0            6.1           2.9            4.7           1.4   
64   65.0            5.6           2.9            3.6           1.3   
66   67.0            5.6           3.0            4.5           1.5   
67   68.0            5.8           2.7            4.1           1.0   
68   69.0            6.2           2.2            4.5           1.5   
69   70.0            5.6           2.5            3.9           1.1   
70   71.0            5.9           3.2            4.8           1.8   
71   72.0            6.1           2.8            4.0           1.3   
72   73.0            6.3           2.5            4.9           1.5   
73   74.0            6.1           2.8            4.7           1.2   
74   75.0            6.4           2.9            4.3           1.3   
78   79.0            6.0           2.9            4.5           1.5   
79   80.0            5.7           2.6            3.5           1.0   
82   83.0            5.8           2.7            3.9           1.2   
83   84.0            6.0           2.7            5.1           1.6   
85   86.0            6.0           3.4            4.5           1.6   
87   88.0            6.3           2.3            4.4           1.3   
88   89.0            5.6           3.0            4.1           1.3   
91   92.0            6.1           3.0            4.6           1.4   
92   93.0            5.8           2.6            4.0           1.2   
94   95.0            5.6           2.7            4.2           1.3   
95   96.0            5.7           3.0            4.2           1.2   
96   97.0            5.7           2.9            4.2           1.3   
97   98.0            6.2           2.9            4.3           1.3   
99  100.0            5.7           2.8            4.1           1.3   

            Species  
51  Iris-versicolor  
55  Iris-versicolor  
56  Iris-versicolor  
61  Iris-versicolor  
62  Iris-versicolor  
63  Iris-versicolor  
64  Iris-versicolor  
66  Iris-versicolor  
67  Iris-versicolor  
68  Iris-versicolor  
69  Iris-versicolor  
70  Iris-versicolor  
71  Iris-versicolor  
72  Iris-versicolor  
73  Iris-versicolor  
74  Iris-versicolor  
78  Iris-versicolor  
79  Iris-versicolor  
82  Iris-versicolor  
83  Iris-versicolor  
85  Iris-versicolor  
87  Iris-versicolor  
88  Iris-versicolor  
91  Iris-versicolor  
92  Iris-versicolor  
94  Iris-versicolor  
95  Iris-versicolor  
96  Iris-versicolor  
97  Iris-versicolor  
99  Iris-versicolor

### 13 Using loc and | to Filter Rows Labeled 'Virginica' or 'Setosa' with 'Petal'WidthCm' Greater Than 2.0

In [136]:
df.loc[((df["Species"] == "Iris-setosa" ) | (df["Species"] == "Iris-virginica")) & (df["PetalWidthCm"] > 2.0)]

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm  \
100  101.0            6.3           3.3            6.0           2.5   
102  103.0            7.1           3.0            5.9           2.1   
104  105.0            6.5           3.0            5.8           2.2   
105  106.0            7.6           3.0            6.6           2.1   
109  110.0            7.2           3.6            6.1           2.5   
112  113.0            6.8           3.0            5.5           2.1   
114  115.0            5.8           2.8            5.1           2.4   
115  116.0            6.4           3.2            5.3           2.3   
117  118.0            7.7           3.8            6.7           2.2   
118  119.0            7.7           2.6            6.9           2.3   
120  121.0            6.9           3.2            5.7           2.3   
124  125.0            6.7           3.3            5.7           2.1   
128  129.0            6.4           2.8            5.6           2.1   
132  133.0            6.4           2.8            5.6           2.2   
135  136.0            7.7           3.0            6.1           2.3   
136  137.0            6.3           3.4            5.6           2.4   
139  140.0            6.9           3.1            5.4           2.1   
140  141.0            6.7           3.1            5.6           2.4   
141  142.0            6.9           3.1            5.1           2.3   
143  144.0            6.8           3.2            5.9           2.3   
144  145.0            6.7           3.3            5.7           2.5   
145  146.0            6.7           3.0            5.2           2.3   
148  149.0            6.2           3.4            5.4           2.3   

            Species  
100  Iris-virginica  
102  Iris-virginica  
104  Iris-virginica  
105  Iris-virginica  
109  Iris-virginica  
112  Iris-virginica  
114  Iris-virginica  
115  Iris-virginica  
117  Iris-virginica  
118  Iris-virginica  
120  Iris-virginica  
124  Iris-virginica  
128  Iris-virginica  
132  Iris-virginica  
135  Iris-virginica  
136  Iris-virginica  
139  Iris-virginica  
140  Iris-virginica  
141  Iris-virginica  
143  Iris-virginica  
144  Iris-virginica  
145  Iris-virginica  
148  Iris-virginica

# **Chipotle Case study**

### 1 Remove the $ symbol from the item_price column and convert it to numeric data

In [167]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 999 entries, 0 to 998
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   order_id            999 non-null    int64  
 1   quantity            999 non-null    int64  
 2   item_name           999 non-null    object 
 3   choice_description  751 non-null    object 
 4   item_price          999 non-null    float64
dtypes: float64(1), int64(2), object(2)
memory usage: 39.2+ KB


In [171]:
df1 = pd.read_csv("chipotle (1).csv")
df1.head(5)

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description item_price  
0                                                NaN     $2.39   
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN     $2.39   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98

In [173]:
df1["item_price"]=df1["item_price"].str.replace("$","").astype("float")
df1["item_price"]

0       2.39
1       3.39
2       3.39
3       2.39
4      16.98
       ...  
994    11.25
995     8.49
996    10.98
997     8.75
998     1.25
Name: item_price, Length: 999, dtype: float64

### 2 How many products cost more than $10?

In [185]:
len(df1["item_price"] > 10)

999

### 3 What is the maximum price of the Chicken Bowl item?

In [204]:
df1.loc[df1["item_name"] == "Chicken Bowl", "item_price"].max()

32.94

### 4 Extract the unique item_name available for less than $5.

In [215]:
df1.loc[df1["item_price"] < 5, "item_name"].unique()


array(['Chips and Fresh Tomato Salsa', 'Izze', 'Nantucket Nectar',
       'Chips and Tomatillo-Green Chili Salsa', 'Side of Chips',
       'Chips and Guacamole', 'Canned Soda', 'Bottled Water',
       'Chips and Tomatillo Green Chili Salsa', 'Chips',
       'Chips and Tomatillo Red Chili Salsa',
       'Chips and Roasted Chili-Corn Salsa',
       'Chips and Roasted Chili Corn Salsa', 'Canned Soft Drink',
       'Chips and Tomatillo-Red Chili Salsa',
       'Chips and Mild Fresh Tomato Salsa'], dtype=object)

### 5 Identify the order_id that has placed the most orders.

In [220]:
df1["order_id"].value_counts().idxmax()

205

### 6 Which is the most ordered item_name?

In [238]:
df1["item_name"].mode()[0]

'Chicken Burrito'

In [234]:
df1["item_name"].value_counts().idxmax()

'Chicken Burrito'

### 7 Find the minimum and maximum price in the item_price column.

In [243]:
df1["item_price"].max()

32.94

In [245]:
df1["item_price"].min()

1.09

### 8 Find the most expensive dish (item_name).

In [256]:
df1.loc[df1["item_price"] == df1["item_price"].max(), "item_name"].iloc[0]


'Chicken Bowl'

### 9 Calculate the total amount for orders where the item_name is Canned Soft Drink and the choice_description is Lemonade.

In [271]:
df1.loc[(df1["item_name"] == "Canned Soft Drink" ) & (df1["choice_description"] == "[Lemonade]"),"item_price" ].sum()

12.5

### 10 Use loc to extract all orders where the item_price is greater than $10 and belongs to the Chicken Bowl.

In [279]:
df1.loc[(df1["item_price"] > 10 ) & (df1["item_name"] == "Chicken Bowl")]

order_id  quantity     item_name  \
4           2         2  Chicken Bowl   
5           3         1  Chicken Bowl   
13          7         1  Chicken Bowl   
42         20         1  Chicken Bowl   
123        54         1  Chicken Bowl   
138        62         1  Chicken Bowl   
140        63         1  Chicken Bowl   
142        64         1  Chicken Bowl   
154        70         2  Chicken Bowl   
176        79         1  Chicken Bowl   
199        89         1  Chicken Bowl   
209        93         1  Chicken Bowl   
211        93         1  Chicken Bowl   
282       124         2  Chicken Bowl   
290       126         1  Chicken Bowl   
291       127         1  Chicken Bowl   
303       132         1  Chicken Bowl   
379       165         1  Chicken Bowl   
407       177         1  Chicken Bowl   
409       178         3  Chicken Bowl   
411       179         1  Chicken Bowl   
415       181         2  Chicken Bowl   
454       195         1  Chicken Bowl   
483       205         1  Chicken Bowl   
490       206         1  Chicken Bowl   
501       210         1  Chicken Bowl   
546       227         1  Chicken Bowl   
548       228         1  Chicken Bowl   
583       239         1  Chicken Bowl   
654       271         2  Chicken Bowl   
668       277         1  Chicken Bowl   
687       284         1  Chicken Bowl   
706       290         1  Chicken Bowl   
719       297         1  Chicken Bowl   
765       316         1  Chicken Bowl   
798       328         1  Chicken Bowl   
817       338         1  Chicken Bowl   
844       347         1  Chicken Bowl   
849       350         1  Chicken Bowl   
871       360         1  Chicken Bowl   
877       362         1  Chicken Bowl   
892       369         1  Chicken Bowl   
915       379         1  Chicken Bowl   
975       400         1  Chicken Bowl   
976       401         2  Chicken Bowl   
993       408         1  Chicken Bowl   
994       408         1  Chicken Bowl   
996       409         1  Chicken Bowl   

                                    choice_description  item_price  
4    [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98  
5    [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...       10.98  
13   [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...       11.25  
42   [Roasted Chili Corn Salsa, [Rice, Black Beans,...       11.25  
123  [Fresh Tomato Salsa, [Guacamole, Cheese, Sour ...       11.25  
138  [Fresh Tomato Salsa, [Rice, Fajita Vegetables,...       11.25  
140  [Tomatillo Green Chili Salsa, [Rice, Sour Crea...       11.25  
142  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...       11.25  
154  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...       17.50  
176  [Roasted Chili Corn Salsa (Medium), [Black Bea...       10.98  
199  [[Roasted Chili Corn Salsa (Medium), Tomatillo...       10.98  
209  [Roasted Chili Corn Salsa, [Fajita Vegetables,...       11.25  
211  [Roasted Chili Corn Salsa, [Fajita Vegetables,...       11.25  
282  [Fresh Tomato Salsa, [Rice, Black Beans, Chees...       17.50  
290  [Roasted Chili Corn Salsa, [Fajita Vegetables,...       11.25  
291  [Tomatillo Green Chili Salsa, [Rice, Sour Crea...       11.25  
303  [Fresh Tomato Salsa, [Rice, Black Beans, Chees...       11.25  
379  [Fresh Tomato Salsa, [Rice, Black Beans, Chees...       11.25  
407  [Tomatillo Red Chili Salsa, [Rice, Black Beans...       11.25  
409  [[Fresh Tomato Salsa (Mild), Tomatillo-Green C...       32.94  
411  [Tomatillo Green Chili Salsa, [Sour Cream, Che...       11.25  
415                        [Tomatillo Red Chili Salsa]       17.50  
454  [Fresh Tomato Salsa, [Rice, Black Beans, Chees...       11.25  
483  [Fresh Tomato Salsa, [Rice, Black Beans, Chees...       11.25  
490  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...       11.25  
501  [Roasted Chili Corn Salsa, [Fajita Vegetables,...       11.25  
546  [Roasted Chili Corn Salsa, [Rice, Pinto Beans,...       11.25  
548  [Roasted Chili Corn Salsa (Medium), [Black Bea...       10

### 11  Use iloc to extract the first 10 rows of the dataset and display only order_id, item_name, and item_price.

In [284]:
df1.iloc[:10,[0,2,4]]

order_id                              item_name  item_price
0         1           Chips and Fresh Tomato Salsa        2.39
1         1                                   Izze        3.39
2         1                       Nantucket Nectar        3.39
3         1  Chips and Tomatillo-Green Chili Salsa        2.39
4         2                           Chicken Bowl       16.98
5         3                           Chicken Bowl       10.98
6         3                          Side of Chips        1.69
7         4                          Steak Burrito       11.75
8         4                       Steak Soft Tacos        9.25
9         5                          Steak Burrito        9.25

### 12 Find the top 3 most ordered items

In [293]:
df1["item_name"].value_counts()[0:3]

item_name
Chicken Burrito        125
Chicken Bowl           119
Chips and Guacamole    110
Name: count, dtype: int64

### 13  Extract all unique choice_description for items costing more than $15

In [299]:
df1.loc[df1["item_price"] > 15,"choice_description"]

4      [Tomatillo-Red Chili Salsa (Hot), [Black Beans...
135    [Tomatillo Green Chili Salsa, [Sour Cream, Che...
148    [Tomatillo-Red Chili Salsa (Hot), [Rice, Chees...
150    [Tomatillo Red Chili Salsa, [Rice, Black Beans...
154    [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...
213    [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...
217    [Tomatillo Red Chili Salsa, [Cheese, Sour Crea...
241    [Tomatillo Red Chili Salsa, [Rice, Black Beans...
281    [Tomatillo Red Chili Salsa, [Black Beans, Chee...
282    [Fresh Tomato Salsa, [Rice, Black Beans, Chees...
315              [Fresh Tomato Salsa, Fajita Vegetables]
353    [Fresh Tomato (Mild), [Lettuce, Guacamole, Ric...
354                     [Fresh Tomato Salsa, Sour Cream]
409    [[Fresh Tomato Salsa (Mild), Tomatillo-Green C...
415                          [Tomatillo Red Chili Salsa]
426          [Fresh Tomato Salsa, [Sour Cream, Lettuce]]
445    [Braised Carnitas, Pinto Beans, [Sour Cream, C...
506    [Roasted Chili Corn Sals

### 14 Find the total quantity of Veggie Salad Bowl sold.

In [304]:
df1.loc[df1["item_name"] == "Veggie Salad Bowl","quantity"].sum()

5

### 15 Find the total revenue generated for each item_name and list the top 5 unique items by revenue

In [317]:
revenue_per_item = df1.groupby("item_name")["item_price"].sum().sort_values(ascending=False).head(5)
revenue_per_item

item_name
Chicken Bowl           1213.86
Chicken Burrito        1205.74
Steak Burrito           961.59
Chips and Guacamole     485.74
Steak Bowl              402.07
Name: item_price, dtype: float64

In [321]:
top5 = (
    df1.pivot_table(values="item_price", index="item_name", aggfunc="sum")
       .sort_values(by="item_price", ascending=False)
       .head(5)
)
print(top5)

                     item_price
item_name                      
Chicken Bowl            1213.86
Chicken Burrito         1205.74
Steak Burrito            961.59
Chips and Guacamole      485.74
Steak Bowl               402.07
